data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionService.java

```diff
@@ -76,6 +79,55 @@ public Object coercePrimaryTerminalNode(final TerminalNode node, final Event eve
                     return longValue;
                 }
                 return Integer.valueOf(nodeStringValue);
+            case DataPrepperExpressionParser.SetInitializer:
+                String[] setMembers = nodeStringValue.trim().substring(1,nodeStringValue.length()-1).split(",");
+                int stringMemberCount = 0;
+                boolean floatMember = false;
+                for (int i= 0; i < setMembers.length; i++) {
+                    String s = setMembers[i].trim();
+                    if (s.length() > 1 && s.charAt(0) == '"' && s.charAt(s.length()-1) == '"') {
+                        stringMemberCount++;
+                    } else if (s.contains(".")) {
+                        floatMember = true;
+                    }
+                }
+                if (stringMemberCount > 0 && stringMemberCount != setMembers.length) {
+                    throw new RuntimeException("All set members should be of same type");
+                }
+                if (stringMemberCount == setMembers.length) {
```

Comment:
Casting the types is something we do elsewhere for other literals right? We should remove this code in favor of using the code that supports existing literals.

Reply:
The code doesn't get called for each of the set members. Let me know if I am missing something.